ts-rs/src/lib.rs

```diff
@@ -289,11 +290,34 @@ pub trait TS {
         panic!("{} cannot be flattened", Self::name())
     }
 
-    /// Information about types this type depends on.
-    /// This is used for resolving imports when exporting to a file.
+    fn dependency_types() -> impl TypeList
+        where
+            Self: 'static { () }
+
     fn dependencies() -> Vec<Dependency>
-    where
-        Self: 'static;
+        where
+            Self: 'static;
+
+    //fn dependencies() -> Vec<Dependency> where Self: 'static {
+    //    use crate::typelist::{TypeList, TypeVisitor};
+    //
+    //    let mut deps: Vec<Dependency> = vec![];
+    //    struct Visit<'a>(&'a mut Vec<Dependency>);
+    //    impl<'a> TypeVisitor for Visit<'a> {
+    //        fn visit<T: TS + 'static + ?Sized>(&mut self) {
+    //            if let Some(dep) = Dependency::from_ty::<T>() {
+    //                self.0.push(dep);
+    //            }
+    //        }
+    //    }
+    //    Self::dependency_types().for_each(&mut Visit(&mut deps));
+    //
+    //    deps
+    //}
```

Comment:
This is how the old `dependencies()` method could be implemented using the new `dependency_types()` method